soc/arm/atmel_sam/common/atmel_sam_dt.h

```diff
@@ -43,4 +45,17 @@
 		ATMEL_SAM_PIN_PERIPH(inst, idx) << 16		\
 	}
 
+/* Devicetree related macros related to clock */
+#ifndef CONFIG_SOC_ATMEL_SAM_MDIV
+#define CONFIG_SOC_ATMEL_SAM_MDIV 1
```

Comment:
Let's not define CONFIG_SOC_ATMEL_SAM_MDIV, just change the code below to be:


```
#ifdef CONFIG_SOC_ATMEL_SAM_MDIV
#define SOC_ATMEL_SAM_MCK_FREQ_HZ \
		(SOC_ATMEL_SAM_HCLK_FREQ_HZ / CONFIG_SOC_ATMEL_SAM_MDIV)
#else
#define SOC_ATMEL_SAM_MCK_FREQ_HZ SOC_ATMEL_SAM_HCLK_FREQ_HZ
#endif

Reply:
In fact, I changed the name because SAME70 is CONFIG_SOC_ATMEL_SAME70_MDIV, SAMV71 is CONFIG_SOC_ATMEL_SAMV71_MDIV and SAM4xxx don't have it. I tried to keep it simple but can rollback, not big deal.